### PR TITLE
Serverless serialize in parallel

### PR DESCRIPTION
**Pull request by ogaca-dd**
### What does this PR do?

This PR adds the support of the parallel serialization for `ServerlessDemultiplexer`.

### Motivation

Having `ServerlessDemultiplexer` behaves the same as `AgentDemultiplexer` makes future development easier.

### Describe how to test/QA your changes

Check series can be sent with the serverless agent.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/aggregator/demultiplexer.go [hunk 1]
```diff
@@ -835,6 +857,20 @@ func (d *ServerlessDemultiplexer) ForceFlushToSerializer(start time.Time, waitFo
 	d.flushLock.Lock()
 	defer d.flushLock.Unlock()
 
+	// only used when we're using flush/serialize in parallel feature
+	logPayloads := config.Datadog.GetBool("log_payloads")
```
Comment (vickenty) [inline pkg/aggregator/demultiplexer.go:861]: Would it make sense to move this into the `startSendingIterableSeries`? Same code with same config var is used when calling it from the demultiplexer.

File: pkg/aggregator/demultiplexer.go [hunk 2]
```diff
@@ -776,7 +793,12 @@ func InitAndStartServerlessDemultiplexer(domainResolvers map[string]resolver.Dom
 	tagsStore := tags.NewStore(config.Datadog.GetBool("aggregator_use_tags_store"), "timesampler")
 
 	statsdSampler := NewTimeSampler(TimeSamplerID(0), bucketSize, tagsStore)
-	statsdWorker := newTimeSamplerWorker(statsdSampler, DefaultFlushInterval, bufferSize, metricSamplePool, flushAndSerializeInParallel{enabled: false}, tagsStore)
+	flushAndSerializeInParallel := flushAndSerializeInParallel{
+		enabled:     config.Datadog.GetBool("aggregator_flush_metrics_and_serialize_in_parallel") && serializer != nil && serializer.IsIterableSeriesSupported(),
+		bufferSize:  config.Datadog.GetInt("aggregator_flush_metrics_and_serialize_in_parallel_buffer_size"),
+		channelSize: config.Datadog.GetInt("aggregator_flush_metrics_and_serialize_in_parallel_chan_size"),
+	}
```
Comment (vickenty) [inline pkg/aggregator/demultiplexer.go:800]: I think the same code exists already in the aggregator (or is it in demultiplexer now?) init code. Would it make senes to factor it out into a helper?

**Review by vickenty — APPROVED** [review]
Just a couple of nitpicks, otherwise LGTM

File: pkg/aggregator/demultiplexer.go (hunk 2 above)
Comment (ogaca-dd) [reply pkg/aggregator/demultiplexer.go:800]: In order to reduce the code conflicts, I have done the code change in the next PR: https://github.com/DataDog/datadog-agent/pull/11133/commits/85e21ffb90f29f199310c3f024217c84ce5f8373.

File: pkg/aggregator/demultiplexer.go (hunk 1 above)
Comment (ogaca-dd) [reply pkg/aggregator/demultiplexer.go:861]: `logPayloads` is also used later in this function: https://github.com/DataDog/datadog-agent/pull/11113/files/775cf7b897e3fcf8e990071877d953b77147bd99#diff-ad4da2ba7c34444c3d1583595302c3b00123265986f7ae6e6c855045d6e0f8e3L579.

**Review by vickenty — APPROVED** [review]

**Pull request merged**